src/main/java/com/transferwise/openbanking/client/api/registration/RegistrationRequestService.java

```diff
@@ -47,7 +47,8 @@ public ClientRegistrationRequest generateRegistrationRequest(String softwareStat
             .tokenEndpointAuthMethod(clientAuthenticationMethod.getMethodName())
             .idTokenSignedResponseAlg(signingAlgorithm)
             .requestObjectSigningAlg(signingAlgorithm)
-            .redirectUris(List.of(tppConfiguration.getRedirectUrl()));
+            .redirectUris(tppConfiguration.getRedirectUrls())
+            .clientId(aspspDetails.getClientId());
```

Comment:
As far as I understand, when updating a client registration, the client ID needs to be specified in both the request path and the request body.